validation/src/collation.rs

```diff
@@ -331,26 +334,29 @@ impl Externalities {
 	// Performs final checks of validity, producing the outgoing message data.
 	fn final_checks(
 		self,
-		candidate: &CandidateReceipt,
-	) -> Result<OutgoingMessages, Error> {
+		candidate: &CollationInfo,
+	) -> Result<(OutgoingMessages, Balance), Error> {
 		if &self.upward != &candidate.upward_messages {
 			return Err(Error::UpwardMessagesInvalid {
 				expected: candidate.upward_messages.clone(),
 				got: self.upward.clone(),
 			});
 		}
 
+		/*
```

Comment:
this logic is important and should not be omitted

Reply:
For the case of candidates received from other validators this logic is moved to `validate_receipt`. So the logic here should probably be just removed.